pkg/cloud/metadata.go

```diff
@@ -130,10 +132,16 @@ func checkAzureMetadata() bool {
 }
 
 // init registers the built-in cloud detectors
-// maintained the original order: GCP first, then AWS, then Azure
 //
 //nolint:gochecknoinits
 func init() {
+	RegisterDefaultDetectors()
+}
+
+// RegisterDefaultDetectors registers the built-in cloud detectors
+//
+// maintained the order: GCP first, then AWS, then Azure
+func RegisterDefaultDetectors() {
```

Comment:
This can be `registerDefaultDetectors`, no?

Reply:
Thanks I'll keep it public so I can choose maybe to init it only on root command and not by init.
Also, it will be good if I'll move the test code to metadata_test package